Source/WTF/wtf/NativePromise.h

```diff
@@ -273,8 +275,51 @@ class NativePromiseBase : public ThreadSafeRefCounted<NativePromiseBase>  {
 
 class ConvertibleToNativePromise { };
 
-template<typename T>
-class NativePromiseRequest;
+class NativePromiseCallback : public ThreadSafeRefCounted<NativePromiseCallback> {
```

Comment:
The name is not right and is further from what the original name properly implied.

You pass to then/whenSettled a callback, what you get back is a request. It's not a callback.